app/app.js

```diff
@@ -0,0 +1,72 @@
+const express = require('express');
+const config = require("./config");
+const errors = require("./util/errors");
+const nunjucks = require("./nunjucks");
+const knex = require("./knex");
+const app = express();
+
+// shift.conf for nginx sets the x-forward header
+// and this says express can use it
+app.set('trust proxy', true);
+
+// allows ex. res.render('crawl.html');
+nunjucks.express(app);
+
+// modify every request
+app.use(function (req, res, next) {
+  // add these two error shortcuts:
+  res.textError = (msg) => errors.textError(res, msg);
+  res.fieldError = (fields, msg) => errors.fieldError(res, fields, msg);
+  // tbd: the php sets this for every end point.
+  // maybe unneeded with the trust_proxy call above?
+  res.set('Access-Control-Allow-Origin', "*");
+  next()
+});
+
+// handle application/x-www-form-urlencoded and application/json posts
+// ( multipart posts are handled by their individual endpoints )
+app.use(express.urlencoded({extended:false}), express.json());
+
+// each of these is a javascript file
+// containing a get ( or post ) export.
+const endpoints = [
+  "crawl",
+  "delete_event",
+  "events",
+  "ical",
+  "manage_event",
+  "retrieve_event"
+];
+
+// host each of those endpoint files at a php-like url:
+// note: require() is synchronous.
+endpoints.forEach((ep) => {
+  const path = `/api/${ep}.php`;
```

Comment:
Long term, I dont think we should be allowing for .php endpoints. Short term, we could add rewrite rules in NGINX?

Reply:
yeah, good point. i wasn't sure if maybe anyone out there in the world was using the api paths directly... rewriting at the ngnix level would be a good middle ground, i think.

Reply:
Well, really I think the BEST solution is to check the logs. If it's not being hit by anything besides our frontend, then it's just tech debt. Although, since this is a decoupled frontend, we may want a temporary redirect anyway until the change can be implemented on the frontend.